aws/retry/middleware.go

```diff
@@ -173,3 +173,14 @@ func getRetryMetadata(ctx context.Context) (metadata retryMetadata, ok bool) {
 func setRetryMetadata(ctx context.Context, metadata retryMetadata) context.Context {
 	return context.WithValue(ctx, retryMetadataKey{}, metadata)
 }
+
+// AwsRetryMiddlewareConfig interface for retry middleware config
+type AwsRetryMiddlewareConfig interface {
+	GetRetryer() Retryer
+}
+
+// AddRetryMiddlewares adds retry middleware to operation middleware stack
+func AddRetryMiddlewares(stack *smithymiddle.Stack, cfg AwsRetryMiddlewareConfig) {
+	stack.Finalize.Add(NewAttemptMiddleware(cfg.GetRetryer(), http.RequestCloner), smithymiddle.After)
+	stack.Finalize.Add(MetricsHeaderMiddleware{}, smithymiddle.After)
+}
```

Comment:
These probably need to refer to the interface `Retryer` not the concrete implementation. These interfaces would actually not be compatible with API client's `aws.Retryer`.  Most likely this means that the middleware cannot be in the `retryer` pkg.  this probably is the same feedback for signer.

Reply:
In the above case, AwsRetryMiddlewareConfig has GetRetryer() which retry a `retry.Retryer` interface. Should this be a smithyhttp.Retryer interface instead?